ompi/mpiext/shortfloat/c/mpiext_shortfloat_c.h.in

```diff
@@ -15,3 +15,7 @@ OMPI_DECLSPEC extern struct ompi_predefined_datatype_t ompi_mpi_cxx_sfltcplex;
 #define MPIX_SHORT_FLOAT             OMPI_PREDEFINED_GLOBAL(MPI_Datatype, ompi_mpi_short_float)
 #define MPIX_C_SHORT_FLOAT_COMPLEX   OMPI_PREDEFINED_GLOBAL(MPI_Datatype, ompi_mpi_c_short_float_complex)
 #define MPIX_CXX_SHORT_FLOAT_COMPLEX OMPI_PREDEFINED_GLOBAL(MPI_Datatype, ompi_mpi_cxx_sfltcplex)
+
+#if @OMPI_MPIX_SHORT_FLOAT_IS_C_FLOAT16@
+#define MPIX_C_FLOAT16               OMPI_PREDEFINED_GLOBAL(MPI_Datatype, ompi_mpi_short_float)
```

Comment:
What do we do for other types that the compiler may or may not support?  I.e., do we declare them in `mpi.h` only if they exist?  Or do we *always* declare them in `mpi.h` and set them to `MPI_DATATYPE_NULL` if they don't exist / aren't supported by the compiler?

Reply:
In `mpi.h`, datatypes are declared only if the corresponding type are supported by the compiler.

Example:
```
#if OMPI_HAVE_FORTRAN_REAL2
#define MPI_REAL2 OMPI_PREDEFINED_GLOBAL(MPI_Datatype, ompi_mpi_real2)
#endif
```

By doing so, `configure` of MPI applications can detect whether the datatypes are supported.